rend3/src/graph/mod.rs

```diff
@@ -869,17 +869,18 @@ impl RenderPassTargets {
     fn compatible(this: Option<&Self>, other: Option<&Self>) -> bool {
         match (this, other) {
             (Some(this), Some(other)) => {
-                let targets_compatible = this
-                    .targets
-                    .iter()
-                    .zip(other.targets.iter())
-                    .all(|(me, you)| me.color == you.color && me.resolve == you.resolve);
-
-                let depth_compatible = this
-                    .depth_stencil
-                    .as_ref()
-                    .zip(other.depth_stencil.as_ref())
-                    .map_or(true, |(me, you)| me.target == you.target);
+                let targets_compatible = this.targets.len() == other.targets.len()
+                    && this
+                        .targets
+                        .iter()
+                        .zip(other.targets.iter())
+                        .all(|(me, you)| me.color == you.color && me.resolve == you.resolve);
+
+                let depth_compatible = match (&this.depth_stencil, &other.depth_stencil) {
```

Comment:
One nit, we should match on (this.depth_stencil, other.depth_stencil) then in the match arm, have (Some(ref this_depth), Some(ref other_depth))

Reply:
Hmmm :thinking:  I tried this:
```rust
                let depth_compatible = match (this.depth_stencil, other.depth_stencil) {
                    (Some(ref this_depth), Some(ref other_depth)) => this_depth == other_depth,
                    (None, None) => true,
                    _ => false,
                };
 ```
 But I'm getting a "cannot move out of `this.depth_stencil` which is behind a shared reference anyway. It's strange though, I also assumed this would work.